app/services/public_market_creation_service.rb

```diff
@@ -1,26 +1,42 @@
 # frozen_string_literal: true
 
-class PublicMarketCreationService < ApplicationService
+require 'ostruct'
+
+class PublicMarketCreationService < ApplicationServiceObject
   def initialize(editor, params)
+    super()
     @editor = editor
     @params = params
   end
 
-  def call
-    validate_editor!
-    create_public_market!
+  def perform
+    validate_editor
+    return self if failure?
+
+    create_public_market
+    self
   end
```

Comment:
Juste une suggestion de style:

```suggestion
  def perform
    validate_editor
    create_public_market if success?

    self
   end


Ou en early return:

 def perform
  return self if (validate_editor; failure?)
  create_public_market
  self
end

```